RGIS2_MergingSpatialData_part1_Joins.Rmd

```diff
@@ -70,24 +70,13 @@ where:
 * _"id-number"_ is the colum that contains the unique identifier in _worldCountries_, and 
 * _"countryID"_ is the column that contains the unique identifier in _countryData_. 
 
-Then we can just merge worldCountries@data with countryData on those variables, with one small nuance: 
-
-**DO NOT MERGE YOUR DATA USING THE `merge` COMMAND!** 
-
-R keeps track of the relationship between rows in the attribute table an `Spatial*` shapes based on the order of observations in the attribute table. Unfortunately, the `merge` command can shuffle the order of rows in the attribute table, potentially jumbling the association of rows with Point/Line/Polygons.
-
-To avoid this problem, always use two rules when merging:
-* Use `join` from the `plyr` library with the `type="left"` argument
-* Also place the DataFrame associated with your `Spatial*` object in the first position. 
+Then we can just merge worldCountries with countryData on those variables, using the `merge` method in `sp`:
```

Comment:
Question on this: is this keyed to the function getting a Spatial* object as an argument? 

R's willingness to let one library "mask" functions from another has always bothered me, since it can make behavior depend on the order of library imports. I just want to be sure that this is "safe".

Relatedly, while I'm happy to use this if it is safe (seems much better, in fact!), I'd still like an explicit warning here about not using merge(worldCountries@data, countryData) because of the order issue. I've had someone in my department (very smart and savvy) make that mistake, and nearly derailed a full research project.